static/js/filter.js

```diff
@@ -99,13 +131,18 @@
             }
 
             const aliasLabel = alias.querySelector(".relay-email-address-label");
+            
 
             if (addOnDetected) {
                 aliasLabel.addEventListener("blur", updateAliasLabel);
+            }
+
+            if (aliasLabel.dataset.label) {
```

Comment:
This change was necessary because every alias was being added to the `aliasesWithLabelsCollection` array, regardless if they actually had a label value saved.